### ISSUE TITLE
validation results while training the pytorch model do not match paper's results 

### ISSUE THREAD

**Issue by SantoshMadhukar** [guxd/deep-code-search#6]
Hi, I have trained the pytorch model with the given [Google Drive](https://drive.google.com/drive/folders/1GZYLT_lzhlVczXjD6dgwVUvDDPHMB6L7?usp=sharing) preprocessed dataset. These are the following validation logs for every 30 epochs while training.

**batch size is 128**

epoch: 30
ACC=0.3093, MRR=0.3093, MAP=0.3093, nDCG=0.3093
epoch: 60
ACC=0.3685, MRR=0.3685, MAP=0.3685, nDCG=0.3685
epoch: 90
ACC=0.3808, MRR=0.3808, MAP=0.3808, nDCG=0.3808
epoch: 120
ACC=0.415, MRR=0.415, MAP=0.415, nDCG=0.415
epoch: 150
ACC=0.4223, MRR=0.4223, MAP=0.4223, nDCG=0.4223
epoch: 180
ACC=0.4306, MRR=0.4306, MAP=0.4306, nDCG=0.4306
epoch: 210
ACC=0.4516, MRR=0.4516, MAP=0.4516, nDCG=0.4516
epoch: 240
ACC=0.4597, MRR=0.4597, MAP=0.4597, nDCG=0.4597
epoch: 270
ACC=0.4647, MRR=0.4647, MAP=0.4647, nDCG=0.4647
epoch: 300
ACC=0.4686, MRR=0.4686, MAP=0.4686, nDCG=0.4686
epoch: 330
ACC=0.4718, MRR=0.4718, MAP=0.4718, nDCG=0.4718
epoch: 360
ACC=0.4704, MRR=0.4704, MAP=0.4704, nDCG=0.4704
epoch: 390
ACC=0.4643, MRR=0.4643, MAP=0.4643, nDCG=0.4643
epoch: 420
ACC=0.4804, MRR=0.4804, MAP=0.4804, nDCG=0.4804
epoch: 450
ACC=0.4741, MRR=0.4741, MAP=0.4741, nDCG=0.4741

I have got only max Mean Reciprocal Rank of 0.48, But in Deep Code Search Paper the MRR is 0.6 with [same dataset](https://drive.google.com/drive/folders/1GZYLT_lzhlVczXjD6dgwVUvDDPHMB6L7?usp=sharing) for 500 epochs (when trainied with Keras model). Please help me with what went wrong?

**Comment by guxd** [comment]
They are totally different results. The MRR reported in the paper is from the test set from StackOverflow. It is calculated according to table 1. While the MRR during training is calculated by negative examples. The validation MRR is just used for monitoring training and was not reported in the paper.

**Comment by SantoshMadhukar** [comment]
Thanks for clarifying!
